src/android/OpenTokAndroidPlugin.java

```diff
@@ -357,28 +375,50 @@ public void onError(SubscriberKit arg0, OpentokError arg1) {
             Log.e(TAG, "subscriber exception: " + arg1.getMessage() + ", stream id: " + arg0.getStream().getStreamId());
         }
 
+        // listeners
         @Override
-        public void onVideoDisableWarning(SubscriberKit arg0) {
-            // TODO Auto-generated method stub
-
+        public void onVideoDataReceived(SubscriberKit arg0) {
+            triggerJSEvent("subscriberEvents", "videoDataReceived", new JSONObject());
```

Comment:
do we need to create an empty object? Why not create a method similar to `triggerJSEvent` where you can pass in a `string` or `null` value.

Reply:
We could do a generic Object implementation like [here](https://github.com/Mediapioniers/cordova-plugin-opentok/blob/aec14a3e5e9624f9be275a57142ed154730fb8aa/src/android/OpenTokAndroidPlugin.java#L782) with a check for sanitizing the received "object"(basically the try catch block already does this for us, we only need to move the type out of the block to emit just the event).

```java
public void triggerJSEvent(String event, String type, Object data) {
        JSONObject message = new JSONObject();

        try {
            message.put("eventType", type);
            message.put("data", data);
        } catch (JSONException e) {
        }

        PluginResult myResult = new PluginResult(PluginResult.Status.OK, message);
        myResult.setKeepCallback(true);
        myEventListeners.get(event).sendPluginResult(myResult);
}
```

Same sort of implementation should be done on the iOS side aswell.

Reply:
that works! :)